File: ObjectAvoidance/java/src/main/java/elite/math/trig/Arcsine.java
package elite.math.trig;

import elite.math.trig.triangle.Hypotenuse;
import elite.math.trig.triangle.Opposite;

/**
 *
 * @author CFerg (Elite)
 */
public class Arcsine {

    /**
     *
     */
    public double arcsine;
    
    /**
     *
     * @param h
     * @param o
     */
    public Arcsine(Hypotenuse h, Opposite o){
        this.arcsine = (h.hypotenuse / o.opposite);
    }
    
    /**
     *
     * @param sin
     */
    public Arcsine(Sine sin){
        this.arcsine = (1 / sin.sine);
    }
    
    /**
     *
     * @param csc
     */
    public Arcsine(Cosecant csc){
        this.arcsine = csc.cosecant;
    }
}

File: ObjectAvoidance/src/elite/math/trig/Arcsine.kt
package elite.math.trig

import elite.math.trig.triangle.Hypotenuse
import elite.math.trig.triangle.Opposite

/**
 *
 * @author CFerg (Elite)
 */
class Arcsine {

    /**
     *
     */
    var arcsine: Double = 0.toDouble()

    /**
     *
     * @param h
     * @param o
     */
    constructor(h: Hypotenuse, o: Opposite) {
        this.arcsine = h.hypotenuse / o.opposite
    }

    /**
     *
     * @param sin
     */
    constructor(sin: Sine) {
        this.arcsine = 1 / sin.sine
    }

    /**
     *
     * @param csc
     */
    constructor(csc: Cosecant) {
        this.arcsine = csc.cosecant
    }
}